packages/twenty-front/src/modules/object-record/record-relation-card/components/RecordRelationFieldCardContent.tsx

```diff
@@ -109,38 +135,66 @@ export const RecordRelationFieldCardContent = ({
         [relationFieldMetadataItem.name]: null,
       },
     });
+
+    modifyObjectMetadataInCache(entityId, {
+      [fieldName]: (relationRef, { readField }) => {
+        const edges = readField<{ node: Reference }[]>('edges', relationRef);
+
+        if (!edges) {
+          return relationRef;
+        }
+
+        return {
+          ...relationRef,
+          edges: edges.filter(({ node }) => {
+            const id = readField('id', node);
+            return id !== relationRecord.id;
+          }),
+        };
+      },
+    });
   };
 
+  const isOpportunityCompanyRelation =
+    (objectMetadataNameSingular === CoreObjectNameSingular.Opportunity &&
+      relationObjectMetadataNameSingular === CoreObjectNameSingular.Company) ||
+    (objectMetadataNameSingular === CoreObjectNameSingular.Company &&
+      relationObjectMetadataNameSingular ===
+        CoreObjectNameSingular.Opportunity);
+
   return (
     <StyledCardContent isDropdownOpen={isDropdownOpen} divider={divider}>
       <FieldContextProvider>
         <FieldDisplay />
       </FieldContextProvider>
-      <DropdownScope dropdownScopeId={dropdownScopeId}>
-        <Dropdown
-          dropdownId={dropdownScopeId}
-          dropdownPlacement="right-start"
-          clickableComponent={
-            <LightIconButton
-              className="displayOnHover"
-              Icon={IconDotsVertical}
-              accent="tertiary"
-            />
-          }
-          dropdownComponents={
-            <DropdownMenuItemsContainer>
-              <MenuItem
-                LeftIcon={IconUnlink}
-                text="Detach"
-                onClick={handleDetach}
+      {/* TODO: temporary to prevent removing a company from an opportunity */}
+      {isOpportunityCompanyRelation && (
```

Comment:
If we want to prevent detaching a company from an opportunity, shouldn't this be: `!isOpportunityCompanyRelation && ...`?